practice/rain.py

```diff
@@ -1,86 +1,214 @@
-    >>> 'This line is the only one that matters'])
-    ['This is the only one that matters']
-    """
-    TABLE_LINE_CUT_OFF = '------------------------------------------------------------------------------------------------------------------\n'
-    table_start_index = text.index(TABLE_LINE_CUT_OFF) + 1
-    table_as_text_lines = text[table_start_index:]
+    """Cuts lines of text after table line.
+
+    >>> cut_non_data([[1], [2], [3], [4], [5], [6], [7], [8], [9], [10], [11], [12], [13], [14]])
+    [[12], [13], [14]]
+    """
+    table_as_text_lines = text[11:]
     return table_as_text_lines
 
 
 def split_single_line_to_single_row(text):
-    """Splits a single line of text into a row in the form of a list of strings"""
-    return text.split('    ')
+    """Splits a single line of text into a row in the form of a list of strings.
+
+    >>> split_single_line_to_single_row('one   two   three' )
+    ['one', 'two', 'three']
+    >>> split_single_line_to_single_row('one-two-three')
+    ['one', 'two', 'three']
+    """
+    simple_row = text.split('   ')
+    full_date = simple_row[0]
+    split_dates = full_date.split('-')
+    final_row = split_dates + simple_row[1:]
+    return final_row
 
 
 def split_lines_to_rows(text):
-    """Splits all lines of text into several rows in the form of lists of strings
-    >>> split_lines_to_rows([['Charmander Bulbasuar Squirtle'],['Ponya Oddish Magicarp'],['Fire Plant Water']])
-    [['Charmander', 'Bulbasuar', 'Squirtle'],['Ponya', 'Oddish', 'Magicarp'],['Fire', 'Plant', 'Water']]
+    """Splits all lines of text into several rows in the form of lists of strings.
+
+    >>> split_lines_to_rows(['Charmander Bulbasuar Squirtle','Ponya Oddish Magicarp','Fire Plant Water'])
+    [['Charmander Bulbasuar Squirtle'], ['Ponya Oddish Magicarp'], ['Fire Plant Water']]
     """
     rows = [split_single_line_to_single_row(line) for line in text]
     return rows
 
 
 def convert_rows_to_columns(rows):
     """Takes rows as lists of strings, and transforms them into columns as lists of strings
-    >>> [['Charmander', 'Bulbasuar', 'Squirtle'],['Ponya', 'Oddish', 'Magicarp'],['Fire', 'Plant', 'Water']]
-    [['Charmander', 'Ponya'', 'Fire'],['Bulbasuar', 'Oddish', 'Plant'],['Squirtle', 'Magicarp', 'Water']]
+
+    >>> convert_rows_to_columns([['Char', 'Bulb', 'Squirt'],['Ponya', 'Oddish', 'Magicarp'],['Fire', 'Plant', 'Water']])
+    [['Char', 'Ponya', 'Fire'], ['Bulb', 'Oddish', 'Plant'], ['Squirt', 'Magicarp', 'Water']]
     """
-    columns = list(zip(*rows))
+    columns_by_tuple = list(zip(*rows))
+    columns = [list(tup) for tup in columns_by_tuple]
     return columns
 
 
-def get_most_rain_inches(columns):
+def contains_numeral(value):
+    """Filtering function, returns boolean if string is numeric.
+    >>> contains_numeral('13')
+    True
+    >>> contains_numeral('pie')
+    False
+    """
+    try:
+        int(value)
+        return True
+    except ValueError:
+        return False
```

Comment:
This function basically re-implements `str.isdigit()` or `str.isnumeric()`. Have a read through your cheat sheet and start to get familiar with all of the std lib functions. I know there are a lot! I just want to start reminding you.